### PR TITLE
refine and fix full_like en api

### PR DESCRIPTION
**Pull request by cryoco**
1. refine full_like en api
2. fix full_like to support setting dtype to the same as input when dtype=None
image
image

### CODE REVIEW

File: python/paddle/tensor/creation.py [hunk 1]
```diff
@@ -81,10 +88,10 @@ def full_like(input,
     helper = LayerHelper("full_like", **locals())
 
     if dtype is None:
-        dtype = 'float32'
+        dtype = input.dtype
 
-    check_dtype(dtype, 'dtype',
-                ['bool', 'float16', 'float32', 'int32', 'int64'], 'full_like')
+    check_dtype(dtype, 'dtype', ['float16', 'float32', 'int32', 'int64'],
```
Comment (NHZlX) [inline python/paddle/tensor/creation.py]: bool type问题，需要确认下，另外这里缺少float64 的type。

Comment (cryoco) [reply python/paddle/tensor/creation.py]: done.

**Review by NHZlX — DISMISSED** [review]
LGTM

**Review by saxon-zh — DISMISSED** [review]

**Review by swtkiwi — APPROVED** [review]
LGTM

**Pull request merged**